grails-app/services/org/pih/warehouse/data/LoadDataService.groovy

```diff
@@ -0,0 +1,291 @@
+            newItem.put("defaultProductPackageQuantity", currentItem.get("Quantity per Package").toInteger());
+            newItem.put("defaultProductPackagePrice", currentItem.get("Package price"));
+
+            csvItems.set(i, newItem);
+        }
+
+        ImportDataCommand command = new ImportDataCommand();
+
+        command.setData(csvItems);
+
+        productSupplierDataService.validate(command)
+        productSupplierDataService.process(command)
+
+        csvReader.close();
+    }
+
+
+    def importInventory(URL csvURL) {
+        CSVMapReader csvReader = new CSVMapReader(csvURL.newInputStream().newReader());
+
+        csvReader.eachLine { Map attr ->
+            Product product = Product.findByProductCode(attr["Product code"]);
+
+            InventoryItem inventoryItem = new InventoryItem(
+                    product: product,
+                    lotNumber: attr["Lot number"],
+                    expirationDate: attr["Expiration date"],
+                    comments: attr["Comment"]
+            );
+
+            inventoryItem.save();
+        }
+
+        csvReader.close();
+    }
+
+    def importInventoryLevels(URL csvURL) {
+        CSVMapReader csvReader = new CSVMapReader(csvURL.newInputStream().newReader());
+
+        csvReader.eachLine { Map<String, String> attr ->
+            Product product = Product.findByProductCode(attr["Product code"]);
+
+            Location preferredBinLocation = Location.findByName(attr["Preferred bin location"])
+
+            InventoryLevel inventoryLevel = new InventoryLevel(
+                    product: product,
+                    status: attr["status"],
+                    preferredBinLocation: preferredBinLocation,
+                    minQuantity: attr["Min quantity"],
+                    maxQuantity: attr["Max quantity"],
+                    reorderQuantity: attr["Reorder quantity"],
+                    expectedLeadTimeDays: attr["Expected Lead Time Days"],
+                    replenishmentPeriodDays: attr["Replenishment Period Days"]
+            )
+
+            inventoryLevel.save(failOnError: true);
+        }
+
+        csvReader.close();
+    }
+
+    def importUsers(URL csvURL) {
+        CSVMapReader csvReader = new CSVMapReader(csvURL.newInputStream().newReader());
+
+        csvReader.eachLine { Map<String, String> attr ->
+            StringTokenizer tokenizer = new StringTokenizer(attr.roleType, ",");
+            Set<RoleType> roleTypes = new HashSet<RoleType>();
+
+            while (tokenizer.hasMoreElements()) {
+                roleTypes.add(RoleType.valueOf(tokenizer.nextElement() as String));
+            }
+
+            Set<Role> userRoles = new HashSet<Role>();
+
+            for (RoleType roleType in roleTypes) {
+                Role role = Role.findByRoleType(roleType)
+
+                if (!role) {
+                    role = new Role(
+                            roleType: roleType,
+                            name: roleType.toString(),
```

Comment:
You should use roleType.name() instead of roleType.toString() to prevent any weirdness with the implicit toString().

https://stackoverflow.com/questions/6667243/using-enum-values-as-string-literals

However, I don't think we want to be creating new system roles during import unless it's explicitly stated in the requirements. I'd rather throw an exception stating that a role with role type doesn't exist. There's also a case where one role type might be mapped to multiple roles. Not sure how we want to handle that. Probably throw an exception for that as well. 
 
Note, once you add a role to the userRoles set, nothing else happens so all of this role / user role logic is basically a no-op as the user roles won't be persisted to the database.

Instead, you probably want to use user.addToUserRoles(). 
```
for (RoleType roleType in roleTypes) {
    Role role = Role.findByRoleType(roleType)
    if (!role) throw new IllegalArgumentException("Role for role type ${roleType.name()} does not exist")
    user.addToUserRoles(new UserRole(user: user, role: role))
}
```
One caveat is that we probably need to save the user before we process the roles.

Reply:
Now null role throws exception. I'm not sure what you mean by no-op - I'm assigning userRoles set to roles property and it works just fine - user and corresponding user_role records are created